### PR TITLE
Upgrade cljdoc dependencies

### LINKED ISSUE #328: Aging cljdoc dependencies make maintenance tricky
While working on various formatting issues, because some of cljdoc's library dependencies are getting a bit stale, I found myself getting confused with current docs for these libraries.

This is especially true of libraries that have had breaking changes, like asciidoctorj and flexmark-java.  

Also some libraries have had security fixes, so we should probably keep up.

I have decided to update most deps to current non-test releases and will summarize any necessary adjustments in this issue.  I have started with [deps.edn](https://github.com/cljdoc/cljdoc/blob/master/deps.edn) but will also review:
1. [modules/deploy/deps.edn](https://github.com/cljdoc/cljdoc/blob/master/modules/deploy/deps.edn)
2. [modules/analysis-runner/deps.edn](https://github.com/cljdoc/cljdoc/blob/master/modules/analysis-runner/deps.edn)
3. [modules/shared-utils/deps.edn](https://github.com/cljdoc/cljdoc/blob/master/modules/shared-utils/deps.edn)
4. [modules/analysis-runner/src/cljdoc/analysis/deps.clj](https://github.com/cljdoc/cljdoc/blob/master/modules/analysis-runner/src/cljdoc/analysis/deps.clj)

If someone knows a good reason NOT to update a dependency, I'm am all ears!  For examples:
1. because the version of JCommander we are using is older than makes sense (2014?), I wonder if there is a good reason not to upgrade.
2. any reason not to reference latest lein, clojure, clojurescript?

### PR DESCRIPTION
**Pull request by lread**
resolves #328

### CODE REVIEW

File: src/cljdoc/util/sqlite_cache.clj [hunk 1]
```diff
@@ -81,7 +84,8 @@
   (lookup [_ k]
     (delay (fetch-item! k (:cache-spec state))))
   (lookup [this k not-found]
-    (delay (or (deref (cache/lookup this k))) not-found))
+    (delay (or (fetch-item! k (:cache-spec state))
+               (derefable? not-found not-found (delay not-found)))))
```
[the remark this answers is not in the document]
Comment (lread) [reply src/cljdoc/util/sqlite_cache.clj]: oopsie, thanks!

File: src/cljdoc/util/sqlite_cache.clj [hunk 2]
```diff
@@ -13,6 +13,9 @@
             [clojure.java.jdbc :as sql])
   (:import (java.time Instant)))
 
+(defn- derefable?[v]
```
[the remark this answers is not in the document]
Comment (lread) [reply src/cljdoc/util/sqlite_cache.clj]: will do

**Comment by lread** [comment]
@martinklepsch, I addressed your review comments, thanks!

Hmm... as part of this PR I could add one more thing.  The cljdoc circleci build could report, where possible, on outdated deps.  No failures, just a report via a simple step.  Do you think that would encourage us to stay more up to date?

**Comment by martinklepsch** [comment]
> No failures, just a report via a simple step. Do you think that would encourage us to stay more up to date?

If it's only printed during CI and doesn't show up on GitHub I'm not sure how many people will see it but if it's a step that is tracked as a "check" on GitHub I can see that being **very useful**.

Screenshot 2019-07-25 at 09 35 16

File: src/cljdoc/util/sqlite_cache.clj [hunk 3]
```diff
@@ -81,7 +84,8 @@
   (lookup [_ k]
     (delay (fetch-item! k (:cache-spec state))))
   (lookup [this k not-found]
-    (delay (or (deref (cache/lookup this k))) not-found))
+    (delay (or (fetch-item! k (:cache-spec state))
+               (if (derefable? not-found) not-found (delay not-found)))))
```
Comment (martinklepsch) [inline src/cljdoc/util/sqlite_cache.clj]: I'm still a little bit confused by this I think.

- It seems as if we will end up with a value that's wrapped in two `delay`s 
- Did you test these things in a REPL / what's your confidence this is correct?

Comment (lread) [reply src/cljdoc/util/sqlite_cache.clj]: Oh gee, you know what? It does look odd. I think I made the correction to your review comment too hastily,  will take more careful look at this.

**Comment by lread** [comment]
> > No failures, just a report via a simple step. Do you think that would encourage us to stay more up to date?
> 
> If it's only printed during CI and doesn't show up on GitHub I'm not sure how many people will see it but if it's a step that is tracked as a "check" on GitHub I can see that being **very useful**.
> 
> Screenshot 2019-07-25 at 09 35 16

Ah, yes, very interesting idea!

File: src/cljdoc/util/sqlite_cache.clj (hunk 3 above)
Comment (lread) [reply src/cljdoc/util/sqlite_cache.clj]: Ok, I think this area deserved some unit tests so added some.
This led me to also find that `memoize/memo-clear!` was not working for clear all case so fixed that too. 
Changes are in this commit: 9c92e1ad0578720e7358ee21e0ec2306d1ba2bec

File: src/cljdoc/util/sqlite_cache.clj [hunk 4]
```diff
@@ -13,11 +13,21 @@
             [clojure.java.jdbc :as sql])
   (:import (java.time Instant)))
 
+(defn instant-now []
+  (Instant/now))
+
+(defn epoch-time-milliseconds []
+  (.toEpochMilli (instant-now)))
```
Comment (martinklepsch) [inline src/cljdoc/util/sqlite_cache.clj]: I personally don't see a lot of value in these kinds of aliases. IMHO they add a layer of indirection that doesn't really add much value. I don't mind keeping this in but just wanted to share a, perhaps differing, viewpoint.

**Comment by martinklepsch** [comment]
Great job on the tests 🙌 Feel free to merge this whenever you feel comfortable. I'd suggest "rebase and merge" so we have all these *beautiful* commits in our Git history ☺️ 

Side note: I'll be offline on Saturday and Sunday so if you think merging this could affect availability  of cljdoc.org you can also wait and we can merge (aka deploy) it together some time on Monday.

**Comment by lread** [comment]
> Great job on the tests 🙌 Feel free to merge this whenever you feel comfortable. I'd suggest "rebase and merge" so we have all these _beautiful_ commits in our Git history ☺️
> 
> Side note: I'll be offline on Saturday and Sunday so if you think merging this could affect availability of cljdoc.org you can also wait and we can merge (i.e. deploy) it together some time on Monday.

Thanks @martinklepsch, this is a pretty big change, so I shall wait until Monday.

File: src/cljdoc/util/sqlite_cache.clj (hunk 4 above)
Comment (lread) [reply src/cljdoc/util/sqlite_cache.clj]: Oh, did not notice this comment until now.
`instant-now` is for test support so we can redef time easily. I will add a comment.
`epoch-time-milliseconds` was not used for test support - I can turf.

**Comment by martinklepsch** [comment]
I can also merge this and watch the deploy go through if you would prefer that

**Comment by lread** [comment]
ya that would be great, thanks! go for it! 🚀

**Pull request merged**